src/app/work-packages/features/work-package/work-package.component.ts

```diff
@@ -77,68 +78,121 @@ export class WorkPackageComponent {
     this.selectedDataset.set(this.datasets().find((d) => d.id === id));
   }
 
+  /**
+   * Remove headers and fix padding of the given base64 encoded key
+   * @param key - the key to be trimmed
+   * @returns the trimmed key
+   */
+  #trimKey(key: string): string {
+    // allow and trim headers and footers for public keys
+    key = key
+      .replace(/-----(BEGIN|END) CRYPT4GH PUBLIC KEY-----/, '')
+      .replace(/^[\s-]+/, '')
+      .replace(/[\s-=]+$/, '');
+    const fix = key.length % 4;
+    if (fix) {
+      key += '='.repeat(4 - fix);
+    }
+    return key;
+  }
+
+  /**
+   * Check whether the given key is a valid public key.
+   * @param key - the key in question
+   * @returns an internal error code
+   */
+  #checkPubKey(key: string): number {
+    key = key.trim();
+    if (!key) {
+      return 1; // key is empty
+    }
```

Comment:
```suggestion
    if (!key) {
      return 1; // key is empty
    }
    key = key.trim();
```
I think the other case is captured in line 114 but the argument might be undefined and in that case, key.trim() would fail I think.

Reply:
The key parameter is guaranteed to be a string. But it could contain only whitespace.

Reply:
I have removed the first check for emptiness now since it's already covered by the second one.

Reply:
If we cannot control the input and/or it is really critical then you shouldn't rely on the typescript checks only, that's true. In this case it's a private method and called with the value of the signal which is a string because it is initialized as an empty string and bound to the value of the input. And the worst thing that could happen is that it throws an error.